### PR TITLE
Feat/input fields

### PR DESCRIPTION
**Pull request by sjnprjl**
# Description

Please include a summary of the changes and the related issue. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by sjnprjl** [comment]
A new route has been added, called /components/, which serves as a documentation route for all the components created thus far. As more components are developed, their documentation can be added to this route to keep everything organized and easily accessible.
Input Documentation
Checkbox docs
Radio component docs

File: src/components/inputs/basic-input.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}) {
+  return (
+    <input
+      type={type}
+      className={classnames("form-control", className, {
+        [onErrorClassName ?? "has-error"]: !!errors[id],
+      })}
+      {...register?.(id, {
+        ...validationSchema,
+        ...(type === "password"
```
Comment (ahnupsingh) [inline src/components/inputs/basic-input.js]: can we make use of default validation like this? `...getDefaultValidationSchema(type)`

File: src/components/inputs/validationSchemas.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  required: "password is required",
+};
+
+export const emailSchema = {
+  pattern: {
+    value: /^[\w-.]+@([\w-]+\.)+[\w-]{2,}$/i,
+    message: "email must be valid.",
+  },
+  required: "email is required",
+};
```
Comment (ahnupsingh) [inline src/components/inputs/validationSchemas.js:15]: add a function here 
```
getDefaultValidationSchema(type){
    return schema based upon type
}
```

File: src/components/inputs/basic-input.js (hunk 1 above)
Comment (sjnprjl) [reply src/components/inputs/basic-input.js]: Created a `getDefaultValidationSchema(type)` factory function

File: src/components/inputs/validationSchemas.js (hunk 2 above)
Comment (sjnprjl) [reply src/components/inputs/validationSchemas.js:15]: Here are the changes:
https://github.com/ahnupsingh/react-crud/pull/18/commits/2a3c6359cbf98f57f1f38db0711da6c2b24fa283
https://github.com/ahnupsingh/react-crud/pull/18/commits/82adb3bed2c59705bf338a86f979ad162595b605

Comment (sjnprjl) [reply src/components/inputs/validationSchemas.js:15]: @ahnupsingh Please review it.

**Pull request merged**